pip_audit/_subprocess.py

```diff
@@ -54,11 +53,14 @@ def run(args: Sequence[str], *, log_stdout: bool = False, state: AuditState = Au
         terminated = process.poll() is not None
         # NOTE(ww): Buffer size chosen arbitrarily here and below.
         stdout += process.stdout.read(4096)  # type: ignore
-        stderr += process.stderr.read(4096)  # type: ignore
         state.update_state(
             f"Running {pretty_args}", stdout.decode(errors="replace") if log_stdout else None
         )
 
+    # NOTE(alex): For reasons I'm unsure about, reading the stderr stream in
+    # real time seems to interfere with stdout and cause us to read nothing.
+    # Let's wait until the process is terminated before reading stderr.
+    stderr = process.stderr.read()  # type: ignore
```

Comment:
Hmm, I'm a little worried this might cause deadlocks in some (unlikely) conditions: I can imagine a case where `pip` writes to `stderr` before `stdout` and blocks waiting for the parent's corresponding read, meaning that `process.poll()` never sees a process termination (and `stdout.read(bufsize)` always results in an empty read).

Specifically, that should only happen if `pip` writes enough to `stderr` to saturate the underlying pipe buffer. But that _might_ happen, so it's probably worth root causing what's going on here instead of deferring the `stderr` read.